### ISSUE TITLE
Duplicate order reference for different order 1/2

### ISSUE THREAD

**Issue by MatShir** [PrestaShop/PrestaShop#20641]
#### Describe the bug

The function orderExists() from Cart.php has been modified in order to use only the cache when the order exists, avoiding the issue with duplicated orders.
**Screenshots**

**Comment by SimonGrn** [comment]
Is this fixed by the linked PR ?

**Comment by Hlavtox** [comment]
Duplicate references are nothing new, huh? It happened to me on 1.6. Solved it like this.

```
public static function generateReference() {  
 
  $reference = strtoupper(Tools::passwdGen(9, 'NO_NUMERIC'));
        
  if (Db::getInstance()->getValue('SELECT reference FROM ' . _DB_PREFIX_ . 'orders WHERE reference = \'' . $reference . '\';')) {
    return Order::generateReference();
  } else {
    return $reference;
  }
		
}
```

**Comment by MatShir** [comment]
Closed by #19139
